### ISSUE TITLE
js/window.location.hostname returns blank string resulting in malfomed websocket url

### ISSUE THREAD

**Issue by martinklepsch** [adzerk-oss/boot-reload#92]
When determining the websocket host in `resolve-url` `(.-hostname (.-location js/window))` returns a blank string for `file://` URLs — I think we should check that the return value is a non-blank string in this case.

We could also consider adding a fallback to `"localhost"` but I'm not fully convinced this is a good idea since it might hide an actual issue. Maybe we can combine the fallback with a warning as well.

``` clojure
(defn resolve-url [url ws-host]
  (let [passed-uri (Uri. url)
        protocol   (.getScheme passed-uri)
        host       (or ws-host
                       (.-hostname (.-location js/window))
                       (do (js/console.warn "Both :ws-host and window.location.hostname are empty."
                                            "This might happen if you are accessing the files directly instead of over http."
                                            "You should probably set :ws-host manually.")
                           nil))
        port       (.getPort passed-uri)]
    (str protocol "://" host ":" port)))

```

**Comment by Deraen** [comment]
Please try https://github.com/adzerk-oss/boot-reload/pull/98

**Comment by Deraen** [comment]
Tested this my self with `file:` URI and this is fixed by #98.
